### PR TITLE
MOD-732 Optimistic mutation: Create new profile

### PR DESCRIPTION
**Pull request by xwdeng**
---
### Summary
Implemented "optimistic mutation" when creating new profile. 
I didn't directly utilize React Query for this implementation. Instead, I introduced an additional layer within the NewProfile component to enhance the user experience.

*Provide a more detailed introduction to the changes in this PR, why they are needed, and what they aim to achieve.*
image

image

### Issues
*Link to the issue

### Type of Change
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update

### Checklist
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published.

### Test
*How did you test your change?

### Additional Context
*Add any other context or screenshots about the pull request

### CODE REVIEW

File: src/decideid_assets/src/components/auth/new_profile/NewProfile.tsx [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
+  useEffect(() => {
+    if (!isLoading) {
+      if (profile?.message == null && profile?.email != null) {
+        navigate('/app');
+      }
+    }
+  }, [isLoading, profile, navigate]);
+
+  return <>
+    { profile?.message ? (
```
Comment (VitalyGitDev) [inline src/decideid_assets/src/components/auth/new_profile/NewProfile.tsx:27]: what is `Message` related to `Profile` ?

File: src/decideid_assets/src/components/auth/new_profile/NewProfile.tsx [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
   const handleCreateProfile = async () => {
     const newProfile = { firstName, lastName, email };
-    await createProfile(newProfile);
+    createProfile(newProfile);
   };
 
-  return (
+  useEffect(() => {
+    if (!isLoading) {
+      if (profile?.message == null && profile?.email != null) {
```
Comment (VitalyGitDev) [inline src/decideid_assets/src/components/auth/new_profile/NewProfile.tsx:20]: I believe its not proper tracking, you should have something like `profile.isLoaded`

File: src/decideid_assets/src/components/auth/new_profile/NewProfile.tsx [hunk 3]
```diff
@@ -1,18 +1,34 @@
 import React, { useState } from "react";
 import { useProfile } from "../../../utils";
+import { useNavigate } from "react-router-dom";
+import { useEffect } from "react";
 
 export default function NewProfile() {
+  const navigate = useNavigate();
+  const { createProfile, profile, isLoading } = useProfile();
```
Comment (VitalyGitDev) [inline src/decideid_assets/src/components/auth/new_profile/NewProfile.tsx:8]: If we are using global StateManagement on FE - why do we need such hook ? I believe we can have just `init()` method which preloads minimal bunch of necessary data...

File: src/decideid_assets/src/components/auth/new_profile/NewProfile.tsx (hunk 2 above)
Comment (xwdeng) [reply src/decideid_assets/src/components/auth/new_profile/NewProfile.tsx:20]: kind of get your point. em.. to reducing confusing, I am going to use the `status` defined by useQuery:
```status: String
Will be:
loading if there's no cached data and no query attempt was finished yet.
error if the query attempt resulted in an error. The corresponding error property has the error received from the attempted fetch
success if the query has received a response with no errors and is ready to display its data. The corresponding data property on the query is the data received from the successful fetch or if the query's enabled property is set to false and has not been fetched yet data is the first initialData supplied to the query on initialization.```

File: src/decideid_assets/src/components/auth/new_profile/NewProfile.tsx (hunk 1 above)
Comment (xwdeng) [reply src/decideid_assets/src/components/auth/new_profile/NewProfile.tsx:27]: Thanks. You are correct. Let me modify this.

**Review by VitalyGitDev — APPROVED** [review]

**Pull request closed without merging**